lib/api/controllers/adminController.js

```diff
@@ -23,28 +23,203 @@
 
 const
   Bluebird = require('bluebird'),
+  {
+    BadRequestError,
+    KuzzleInternalError
+  } = require('kuzzle-common-objects').errors,
+  Request = require('kuzzle-common-objects').Request,
+  {
+    assertArgsHasAttribute
+  } = require('../../util/requestAssertions'),
   path = require('path'),
   fs = require('fs-extra'),
   os = require('os'),
-  Request = require('kuzzle-common-objects').Request,
   moment = require('moment'),
   dumpme = require('dumpme'),
   zlib = require('zlib'),
   glob = require('glob');
 
 let
   _lock = false,
-  _kuzzle;
+  _shutdown = false;
+
+/**
+ * @class AdminController
+ * @param {Kuzzle} kuzzle
+ */
+class AdminController {
+  constructor(kuzzle) {
+    this.kuzzle = kuzzle;
+  }
+
+  /**
+   * Reset the internal storage components (internalEngine index, cache and memory storage)
+   */
+  resetKuzzleData () {
+    this.kuzzle.pluginsManager.trigger('log:info', 'Kuzzle reset initiated: this may take a while...');
+
+    return deleteObjects(this.kuzzle, 'user', {})
+      .then(() => this.kuzzle.internalEngine.deleteIndex())
+      .then(() => this.kuzzle.pluginsManager.trigger('log:info', 'Kuzzle internal database deleted'))
+      .then(() => this.kuzzle.services.list.internalCache.flushdb())
+      .then(() => {
+        this.kuzzle.indexCache.remove(this.kuzzle.internalEngine.index);
+        this.kuzzle.pluginsManager.trigger('log:info', 'Kuzzle internal cache flushed');
+
+        return this.kuzzle.internalEngine.bootstrap.all();
+      });
+  }
+
+  /**
+   * Reset Redis cache
+   */
+  resetCache (request) {
+    assertArgsHasAttribute(request, 'database');
+
+    const
+      database = request.input.args.database,
+      cacheEngine = this.kuzzle.services.list[database];
+
+    if (! (cacheEngine !== undefined && typeof cacheEngine.flushdb === 'function')) {
+      throw new BadRequestError(`Database ${database} not found`);
+    }
+
+    return new Bluebird((resolve, reject) => cacheEngine.flushdb(error => error ? reject(error) : resolve()));
+  }
+
+  /**
+   * Reset all roles, profiles and users
+   */
+  resetSecurity () {
+    const options = {
+      refresh: 'wait_for'
+    };
+
+    return deleteObjects(this.kuzzle, 'user', options)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Add a semicolon at the end of this statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ASemicolon)